rfcs/0107-version-normalization.md

```diff
@@ -0,0 +1,211 @@
+---
+feature: nixpkgs_version_normalization
+start-date: 2021-09-24
+author: Anderson Torres
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Normalize the `version` attribute used in Nixpkgs' expresions.
+
+# Motivation
+[motivation]: #motivation
+
+Nowadays, the most commonly used format for the `pname` and `version` attributes
+along the Nixpkgs' expressions is:
+
+- For stable releases:
+  - `pname` reflects the software name;
+  - `version` reflects the released version.
+- For unstable releases:
+  - `pname` reflects the software name;
+  - `version` is a string following the format `unstable-YYYY-MM-DD`, where
+    `YYYY-MM-DD` denotes the date when the code was generated.
+
+This is a simple and easy-to-understand format. Nonetheless, there are some
+problems with it.
+
+First, it does not map very well with the Nix function
+`builtins.parseDrvName`:
+
+```example
+# expected: { name = "mpv"; version = "0.35.15"; }
+nix-repl> builtins.parseDrvName "mpv-0.35.15"
+builtins.parseDrvName "mpv-0.35.15"
+{ name = "mpv"; version = "0.35.15"; }
+
+# expected: { name = "mpv"; version = "unstable-2021-05-03"; }
+nix-repl> builtins.parseDrvName "mpv-unstable-2021-05-03"
+builtins.parseDrvName "mpv-unstable-2021-05-03"
+{ name = "mpv-unstable"; version = "2021-05-03"; }
+```
+
+It happens because the `version` attribute in the set returned by
+`builtins.parseDrvName` always starts with a digit. This is a deliberate design
+decision of Nix specification, and as such it should not be regarded as a "bug";
```

Comment:
How exactly? Can you explain better?

Reply:
(I also did not understand how this would work)

Reply:
I needed to think what I meant, too. If we could change how ``builtins.parseDrvName`` works and just treats unstable as part of the version number we would be done in no time.

Reply:
Well, it would be nice if the new work played nice with `builtins.compareVersions` for conditional inclusion/exclusion of patches and similar purposes…